### PR TITLE
GIT-963_Adding listMetadata functionality

### LINKED ISSUE #963: [Connector] Add file metadata fetch support to msazurestorage connector
### Problem

We have introduced download blob support to msazurestorage via [1].
While we are doing the feature, we noticed that we can also retrieve file metadata details.
We need a new connector operation to fetch these metadata details 

[1] https://github.com/wso2-extensions/esb-connector-msazurestorage/pull/15

### Solution

Add new operation to fetch blob metadata

### Affected Component

MI

### Version

_No response_

### Implementation

_No response_

### Related Issues

_No response_

### Suggested Labels

_No response_

### PR DESCRIPTION
**Pull request by pabloa2022**
Resolves https://github.com/wso2/api-manager/issues/963

### CODE REVIEW

File: src/main/resources/metadata/component.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Copyright (c) 2018, WSO2 LLc. (http://www.wso2.org) All Rights Reserved.
```
Comment (arunans23) [inline src/main/resources/metadata/component.xml]: Let's change the year here.

File: src/main/resources/metadata/listMetadata.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations
+   under the License.
+-->
+<template xmlns="http://ws.apache.org/ns/synapse" name="listMetadata">
+    <parameter defaultValue="" isMandatory="false" name="containerName"/>
```
Comment (SanojPunchihewa) [inline src/main/resources/metadata/listMetadata.xml]: Aren't these two parameters mandatory?

File: src/main/resources/metadata/component.xml (hunk 1 above)
Comment (pabloa2022) [reply src/main/resources/metadata/component.xml]: Year changed. Thanks

File: src/main/resources/metadata/listMetadata.xml (hunk 2 above)
Comment (pabloa2022) [reply src/main/resources/metadata/listMetadata.xml]: Hi Sanoj, you are right. Fixed.
Thanks!

**Review by arunans23 — DISMISSED** [review]

File: src/main/java/org/wso2/carbon/connector/azure/storage/ListMetadata.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/azure/storage/ListMetadata.java]:
```suggestion
 *  WSO2 LLC. licenses this file to you under the Apache License,
```

File: src/main/resources/metadata/component.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Copyright (c) 2022, WSO2 LLc. (http://www.wso2.org) All Rights Reserved.
+
+   WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (SanojPunchihewa) [inline src/main/resources/metadata/component.xml]:
```suggestion
   WSO2 LLC. licenses this file to you under the Apache License,
```

File: src/main/resources/metadata/listMetadata.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<!--
+   Copyright (c) 2022, WSO2 LLc. (http://www.wso2.org) All Rights Reserved.
```
Comment (SanojPunchihewa) [inline src/main/resources/metadata/listMetadata.xml]: Please check other places as well

```suggestion
   Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

**Review by SanojPunchihewa — APPROVED** [review]

**Review by arunans23 — APPROVED** [review]

**Pull request merged**